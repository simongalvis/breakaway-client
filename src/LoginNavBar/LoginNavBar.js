import React from 'react';
import { Link } from 'react-router-dom';
import './LoginNavBar.css';
import logo from '../images/breakaway-logo.png'



class LoginNavBar extends React.Component{



    render(){
        return(
            <header>
                <Link to='/'>
                    <img src={logo} alt='Blue breakaway logo' id='main-logo'/>
                </Link>
                <Link to="/signup-page">
                    <button className='nav-button' id='signup-button' value='signup'>Sign Up</button>
                </Link>
                
                
            </header>
            
        )
    }
}


export default LoginNavBar;