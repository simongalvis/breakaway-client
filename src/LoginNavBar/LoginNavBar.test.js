import React from 'react'
import ReactDOM from 'react-dom'
import LoginNavBar from './LoginNavBar'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer';

describe(`LoginNavBar component`, () =>{
  it(`renders without crashing`, () =>{
    const div = document.createElement('div');
  
    ReactDOM.render(<BrowserRouter><LoginNavBar/></BrowserRouter>, div);
  
    ReactDOM.unmountComponentAtNode(div);
  })
  it(`Renders UI as expected`, () =>{
    const tree = renderer
      .create(<BrowserRouter><LoginNavBar/></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})
