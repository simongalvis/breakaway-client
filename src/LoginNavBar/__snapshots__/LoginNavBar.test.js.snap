// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginNavBar component Renders UI as expected 1`] = `
<header>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      alt="Blue breakaway logo"
      id="main-logo"
      src="breakaway-logo.png"
    />
  </a>
  <a
    href="/signup-page"
    onClick={[Function]}
  >
    <button
      className="nav-button"
      id="signup-button"
      value="signup"
    >
      Sign Up
    </button>
  </a>
</header>
`;
